### ISSUE TITLE
Consider communicating Lua data to visualization

### ISSUE THREAD

**Issue by LinearZoetrope** [SCAII/SCAII#109]
This has two related pieces:

1. It would be useful to communicate **all** Lua globals (#102) to the visualization in a debug case (relies on #100), so we can see values changing frame to frame to debug inconsistencies.
2. It would also be useful to communicate **specific** Lua globals all the time if they're important to the scenario.

**Comment by jedirv** [comment]
Here's my best guess at how this might be done (sans some details):

create a struct called DebugInfo which wraps a HashMap

to register_world_components() add...

     world.register::<DebugInfo>();

then in LuaSystemData we add

    debug_info: WriteStorage<'a, DebugInfo>,

and in RenderSystemData we add 
      debug_info: ReadStorage<'a, DebugInfo>,

Then when lua stuff is happening we save stuff into DebugInfo for the game-side entity and then when we render into ScaiiEntity in RenderSystem, we can just extract all the _debug* keys and values from DebugInfo and pass into string_metadata map.

Am I close?

**Comment by LinearZoetrope** [comment]
No. You're going to want to look at the data stores in `systems/lua/userdata.rs`. I was thinking of adding a `display` HashSet to either that or the `World` that contains a list of keys that will always be sent to the visualization. By default they'd only be sent in a debug mode.

Debug mode would be a protobuf config tweak. The game, if given a `debug_mode` config (from the internal config message, not the main `scaii_defs` one) would set a new **resource** (not component) to true, and if true, would prefix all keys *not* in the `display` HashSet with `debug_` and send them over.

**Comment by jedirv** [comment]
two phases 
- prefix all the info with debug_ and send it over
- add the debug discriminator and choose which

**Comment by jedirv** [comment]
Went ahead and committed minor work done so far into debug-info branch (branched off dev)
